build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```
@@ -21,6 +24,61 @@ plugins {
     id("base")
     id("checkstyle")
     id("codenarc")
+    id("net.ltgt.errorprone")
+}
+
+open class ErrorProneProjectExtension(
+    val disabledChecks: ListProperty<String>
+)
+
+open class ErrorProneSourceSetExtension(
+    val enabled: Property<Boolean>
+)
+
+val errorproneExtension = project.extensions.create<ErrorProneProjectExtension>("errorprone", project.objects.listProperty<String>())
+errorproneExtension.disabledChecks.addAll(
+    // DISCUSS
+    "EqualsGetClass", // Let's agree if we want to adopt Error Prone's idea of valid equals()
+    "JdkObsolete", // Most of the checks are good, but we do not want to replace all LinkedLists without a good reason
+
+    // NEVER
+    "MissingSummary", // We have another mechanism to check Javadocs on public API
+    "InjectOnConstructorOfAbstractClass", // We use abstract injection as a pattern
+    "JavaxInjectOnAbstractMethod", // We use abstract injection as a pattern
+    "JavaUtilDate", // We are fine with using Date
+)
+
+project.plugins.withType<JavaBasePlugin> {
+    project.extensions.getByName<SourceSetContainer>("sourceSets").configureEach {
+        val extension = this.extensions.create<ErrorProneSourceSetExtension>("errorprone", project.objects.property<Boolean>())
+        // Enable it only for the main source set by default, as incremental Groovy
+        // joint-compilation doesn't work with the Error Prone annotation processor
+        extension.enabled.convention(this.name == "main")
+
+        project.dependencies.addProvider(
+            annotationProcessorConfigurationName,
+            extension.enabled.filter { it }.map { "com.google.errorprone:error_prone_core:2.24.1" }
+        )
+
+        project.tasks.named<JavaCompile>(this.compileJavaTaskName) {
+            options.errorprone {
+                isEnabled = extension.enabled
+                checks.set(errorproneExtension.disabledChecks.map {
+                    it.stream().collect(Collectors.toMap(
+                        { check -> check },
+                        { _ -> CheckSeverity.OFF }
+                    ))
```

WDYT of the following?

```suggestion
                    it.associateWith { CheckSeverity.OFF }
```

---

Looks good, thanks.